sor/src/main/java/com/bazaarvoice/emodb/sor/compactioncontrol/DefaultCompactionControlSource.java

```diff
@@ -102,14 +102,14 @@ protected static String zkKey(String id, String dataCenter) {
     }
 
     @VisibleForTesting
-    protected static Map<Pair, StashRunTimeInfo> getStashTimesWithTupleKeys(Map<String, StashRunTimeInfo> stashTimesFromZk) {
-        Map<Pair, StashRunTimeInfo> allStashTimes = Maps.newHashMap();
+    protected static Map<StashTimeKey, StashRunTimeInfo> getStashTimesWithTupleKeys(Map<String, StashRunTimeInfo> stashTimesFromZk) {
+        Map<StashTimeKey, StashRunTimeInfo> allStashTimes = Maps.newHashMap();
         for (Map.Entry<String, StashRunTimeInfo>  stashTimeFromZk : stashTimesFromZk.entrySet()) {
             String[] stashTimeFromZkSplits = stashTimeFromZk.getKey().split(ZK_STRING_DELIMITER);
             if (stashTimeFromZkSplits.length != 2) {
                 throw new IllegalStateException("Zookeeper key can only contain one '@' character. Pattern is ID@datacenter");
             }
-            allStashTimes.put(Pair.of(stashTimeFromZkSplits[0], stashTimeFromZkSplits[1]), stashTimeFromZk.getValue());
+            allStashTimes.put(StashTimeKey.of(stashTimeFromZkSplits[0], stashTimeFromZkSplits[1]), stashTimeFromZk.getValue());
```

Comment:
This is really minor and only a suggestion, but since `StashTimeKey.toString()` generates the "id@datacenter" string, maybe have a `StashTimeKey.fromString(String s)` which reverses this process, basically centralizing the serialization in that one class.